src/python_testing/TC_XX_1_1.py

```diff
@@ -0,0 +1,138 @@
+        if suffix > attribute_id_range_max and suffix < attribute_global_range_min or suffix == attribute_id_invalid_max:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in standard range")
+
+    elif prefix >= manufacturer_code_min_prefix and prefix <= manufacturer_code_max_prefix:
+        if suffix > attribute_id_range_max:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in MC range")
+
+    elif prefix >= test_vendor_min_prefix and prefix <= test_vendor_max_prefix:
+        if suffix >= attribute_id_range_min:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in Test Vendor MC range")
+
+    else:  # prefix >= invalid_vendor_min_prefix and prefix >= invalid_vendor_max_prefix
+        if suffix >= attribute_id_range_min:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in invalid range")
+
+
+def is_command_id_in_valid_range(command_id):
+    prefix, suffix = get_element_prefix_suffix(command_id)
+
+    if prefix == standard_range_prefix:
+        if suffix > command_id_range_max:
+            asserts.fail(f"Invalid command id {command_id:010x} in standard range")
+
+    elif prefix >= manufacturer_code_min_prefix and prefix <= manufacturer_code_max_prefix:
+        if suffix > command_id_range_max:
+            asserts.fail(f"Invalid command id {command_id:010x} in MC range")
+
+    elif prefix >= test_vendor_min_prefix and prefix <= test_vendor_max_prefix:
+        if suffix >= command_id_range_min:
+            asserts.fail(f"Invalid command id {command_id:010x} in Test Vendor MC range")
+
+    else:  # prefix >= invalid_vendor_min_prefix and prefix >= invalid_vendor_max_prefix
+        if suffix >= command_id_range_min:
+            asserts.fail(f"Invalid command id {command_id:010x} in invalid range")
+
+
+class TC_XX_1_1(MatterBaseTest):
+    @async_test_body
+    async def test_xx_1_1(self):
+
+        dev_ctrl = self.default_controller
+
+        self.print_step(1, "Perform a wildcard read of attributes on all endpoints")
+        wildcard_read = (await dev_ctrl.Read(self.dut_node_id, [()]))
+        endpoints_tlv = wildcard_read.tlvAttributes
+
+        self.print_step(2, "Check invalid elements on all clusters, across all endpoints")
+        for endpoint_id, endpoint in endpoints_tlv.items():
+            logging.info(f"Verifying Endpoint: {endpoint_id}")
+
+            for cluster_id, cluster in endpoint.items():
+                logging.info(f"Verifying ClusterID: {cluster_id:#06x}")
+
+                attribute_list = cluster[ATTRIBUTE_LIST_ID]
+                accepted_command_list = cluster[ACCEPTED_COMMAND_LIST_ID]
+                generated_command_list = cluster[GENERATED_COMMAND_LIST_ID]
+
+                # -- Attributes
+                logging.info(f"AttributeList: {attribute_list}")
+                for attribute_id in attribute_list:
+                    is_attribute_id_in_valid_range(attribute_id)
```

Comment:
is_* sounds like a method returning a bool. Should maybe be named `validate_attribute_id_range` or similar

Reply:
Agree, the "is" word does give that impression. So I will update it.

Reply:
This is addressed in the latest commit